File: HRMS/HRMS.Web/Modules/Recruitment/JobOpenings/JobOpeningsForm.cs
using Serenity.ComponentModel;
using HRMS.HR;

namespace HRMS.Recruitment.Forms;

[FormScript("Recruitment.JobOpenings")]
[BasedOnRow(typeof(JobOpeningsRow), CheckNames = true)]
public class JobOpeningsForm
{
    public string Title { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Description { get; set; }
    [LookupEditor(typeof(DepartmentRow))]
    public int DepartmentId { get; set; }
    [LookupEditor(typeof(EmployeeRow))]
    public int HiringManagerId { get; set; }
    public int Status { get; set; }
    [DefaultValue("now")]
    public DateTime CreatedDate { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Recruitment/JobOpeningsForm.ts
import { DateEditor, EnumEditor, initFormType, LookupEditor, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";
import { JobOpeningStatus } from "./JobOpeningStatus";

export interface JobOpeningsForm {
    Title: StringEditor;
    Description: TextAreaEditor;
    DepartmentId: LookupEditor;
    HiringManagerId: LookupEditor;
    Status: EnumEditor;
    CreatedDate: DateEditor;
}

export class JobOpeningsForm extends PrefixedContext {
    static readonly formKey = 'Recruitment.JobOpenings';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!JobOpeningsForm.init) {
            JobOpeningsForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = LookupEditor;
            var w3 = EnumEditor;
            var w4 = DateEditor;

            initFormType(JobOpeningsForm, [
                'Title', w0,
                'Description', w1,
                'DepartmentId', w2,
                'HiringManagerId', w2,
                'Status', w3,
                'CreatedDate', w4
            ]);
        }
    }
}

[JobOpeningStatus]; // referenced types